docs/benchmarking-workflow-dotnet-runtime.md

```diff
@@ -187,6 +200,268 @@ If you are seeing warnings like:
+```
+
+#### Running on Windows
+
+```cmd
+# Set $RunDir to the runtime directory
+$RunDir=""FullPathHere""
+
+# Set the OS, arch, and OSId
+RunOS='windows'
+RunOSId='win'
+RunArch='x64'
+
+# Create mono dotnet
+mkdir -p $RunDir/artifacts/dotnet-mono
+$RunDir/build.sh -subset libs.pretest -configuration release -ci -arch $RunArch -testscope innerloop /p:RuntimeArtifactsPath=$RunDir\artifacts\bin\mono\$RunOS.$RunArch.Release /p:RuntimeFlavor=mono
+xcopy $RunDir\artifacts\bin\runtime\net8.0-$RunOS-Release-$RunArch\ $RunDir\artifacts\bin\testhost\net8.0-$RunOS-Release-$RunArch\shared\Microsoft.NETCore.App\8.0.0\ /e /y
+xcopy $RunDir\artifacts\bin\testhost\net8.0-$RunOS-Release-$RunArch\ $RunDir\artifacts\dotnet-mono\ /e /y
+xcopy $RunDir\artifacts\bin\coreclr\$RunOS.$RunArch.Release\corerun $RunDir\artifacts\dotnet-mono\shared\Microsoft.NETCore.App\8.0.0\corerun /y
+
+# Create Core Root
+$RunDir\src\tests\build.cmd release $RunArch generatelayoutonly /p:LibrariesConfiguration=Release
+
+# Clone performance 
+git clone --branch main --depth 1 --quiet https://github.com/dotnet/performance.git $RunDir\performance
+
+# Use Interpreter
+export MONO_ENV_OPTIONS=""--interpreter""
+
+# One line run:
+python3 $RunDir\performance\scripts\benchmarks_ci.py --csproj $RunDir\performance\src\benchmarks\micro\MicroBenchmarks.csproj --incremental no --architecture $RunArch -f net8.0 --filter '$TestToRun*' --bdn-artifacts $RunDir\artifacts\BenchmarkDotNet.Artifacts --bdn-arguments=""--anyCategories Libraries Runtime  --category-exclusion-filter NoInterpreter NoMono --logBuildOutput --generateBinLog --corerun $RunDir\artifacts\dotnet-mono\shared\Microsoft.NETCore.App\8.0.0\corerun.exe"" 
+
+# Individual Commands:
+# Restore 
+dotnet restore $RunDir\performance\src\benchmarks\micro\MicroBenchmarks.csproj --packages $RunDir\performance\artifacts\packages /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1
+
+# Build
+dotnet build $RunDir\performance\src\benchmarks\micro\MicroBenchmarks.csproj --configuration Release --framework net8.0 --no-restore /p:NuGetPackageRoot=$RunDir\performance\artifacts\packages /p:UseSharedCompilation=false /p:BuildInParallel=false /m:1
+
+# Run
+dotnet run --project $RunDir\performance\src\benchmarks\micro\MicroBenchmarks.csproj --configuration Release --framework net8.0 --no-restore --no-build -- --filter '$TestToRun*' --anyCategories Libraries Runtime "" --category-exclusion-filter NoInterpreter NoMono --logBuildOutput --generateBinLog --corerun $RunDir\artifacts\dotnet-mono\shared\Microsoft.NETCore.App\8.0.0\corerun.exe --artifacts $RunDir\artifacts\BenchmarkDotNet.Artifacts --packages $RunDir\performance\artifacts\packages
+```
+
+### dotnet runtime testing for Mono Default (JIT)
```

Comment:
Appreciate the instructions existing now. Have some feedback now that I've had a chance to try it out.

First, the Windows instructions are "incorrect" in that they are still using bash. They should be updated to ensure they "just work" with a regular Windows command line or powershell environment. There are also some sections (wasm) where no windows instructions exist at all.

Secondly, the instructions here are altogether too verbose. The setup for `RyuJIT` is two-three commands:
```
# in dotnet/runtime-base
.\build.cmd -subset clr+libs -config release

# in dotnet/runtime
.\build.cmd -subset clr+libs -config release

# in dotnet/performance, under the src/benchmarks/micro folder
dotnet run -c Release -f net8.0 --filter $someFilter --coreRun 
C:\Users\tagoo\source\repos\runtime-base\artifacts\bin\testhost\net8.0-windows-Release-x64\shared\Microsoft.NETCore.App\8.0.0\CoreRun.exe
C:\Users\tagoo\source\repos\runtime\artifacts\bin\testhost\net8.0-windows-Release-x64\shared\Microsoft.NETCore.App\8.0.0\CoreRun.exe
```

These two commands give me an end-to-end ready to use setup comparing a baseline and diff build. Compared to Mono, where there are many additional and explicit setup instructions required, it makes Mono overall very difficult to use.

Ideally, we could get mono similarly down to 2-3 commands. I'd imagine a very nice setup would be something like:
```csharp
# in dotnet/runtime-base
.\build.cmd -subset mono+libs -config release

# in dotnet/runtime
.\build.cmd -subset mono+libs -config release

# in dotnet/performance, under the src/benchmarks/micro folder
dotnet run -c Release -f net8.0 --filter $someFilter --mono --coreRun 
C:\Users\tagoo\source\repos\runtime-base\artifacts\bin\testhost\net8.0-windows-Release-x64\shared\Microsoft.NETCore.App\8.0.0\CoreRun.exe
C:\Users\tagoo\source\repos\runtime\artifacts\bin\testhost\net8.0-windows-Release-x64\shared\Microsoft.NETCore.App\8.0.0\CoreRun.exe
```

Being able to simply pass in `--interpreter` to the `dotnet run` command to opt-in to interpreted results would also be beneficial. (Naturally the exact command might differ slightly, but the general point is we should be able to get a usable "sdk" with one command and then pass that down to the performance repo in another).

Reply:
CC. @SamMonoRT

Reply:
Thanks for trying these out @tannergooding, really appreciate the feedback. @cincuranet @LoopedBard3 we can work to try to improve it. @tannergooding, looking forward to the mono wasm-aot steps.

Reply:
@SamMonoRT Sure thing. I think the best approach is to create issue(s) in this repo and we'll work on that.